ginivision/src/main/java/net/gini/android/vision/GiniVision.java

```diff
@@ -0,0 +1,56 @@
+package net.gini.android.vision;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+
+import net.gini.android.vision.analysis.AnalysisActivity;
+import net.gini.android.vision.document.DocumentFactory;
+import net.gini.android.vision.internal.util.ActivityHelper;
+import net.gini.android.vision.internal.util.DeviceHelper;
+import net.gini.android.vision.internal.util.FileImportValidator;
+import net.gini.android.vision.internal.util.IntentHelper;
+import net.gini.android.vision.review.ReviewActivity;
+
+public final class GiniVision {
+
+    public static Intent createIntentForImportedFile(@NonNull final Intent intent,
+            @NonNull final Context context,
+            @NonNull final Class<? extends ReviewActivity> reviewActivityClass,
+            @NonNull final Class<? extends AnalysisActivity> analysisActivityClass)
+            throws ImportedFileValidationException {
+        final Document document = createDocumentForImportedFile(intent, context);
+        final Intent giniVisionIntent;
+        if (document.isReviewable()) {
+            giniVisionIntent = new Intent(context, reviewActivityClass);
+            giniVisionIntent.putExtra(ReviewActivity.EXTRA_IN_DOCUMENT, document);
+            ActivityHelper.setActivityExtra(giniVisionIntent,
+                    ReviewActivity.EXTRA_IN_ANALYSIS_ACTIVITY,
+                    context,
+                    analysisActivityClass);
+        } else {
+            giniVisionIntent = new Intent(context, analysisActivityClass);
+            giniVisionIntent.putExtra(AnalysisActivity.EXTRA_IN_DOCUMENT, document);
+        }
+        return giniVisionIntent;
+    }
+
+    @NonNull
+    public static Document createDocumentForImportedFile(@NonNull final Intent intent,
+            @NonNull final Context context) throws ImportedFileValidationException {
+        final Uri uri = IntentHelper.getUri(intent);
+        if (uri == null) {
+            throw new ImportedFileValidationException("todo description no uri");
```

Comment:
I saw that `ImportedFileValidationException` stack trace is printed in other classes, so maybe we should add a proper description here.

Reply:
Whew, thanks for noticing this!